akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -123,6 +123,21 @@ object ActorMaterializer {
   def create(context: ActorRefFactory): ActorMaterializer =
     apply()(context)
 
+  /**
+   * Java API: Creates an ActorMaterializer that can materialize stream blueprints as running streams.
+   *
+   * The required [[akka.actor.ActorRefFactory]]
+   * (which can be either an [[akka.actor.ActorSystem]] or an [[akka.actor.ActorContext]])
+   * will be used to create these actors, therefore it is *forbidden* to pass this object
+   * to another actor if the factory is an ActorContext.
+   *
+   * The `namePrefix` is used as the first part of the names of the actors running
+   * the processing steps. The default `namePrefix` is `"flow"`. The actor names are built up of
+   * `namePrefix-flowNumber-flowStepNumber-stepName`.
+   */
+  def create(context: ActorRefFactory, namePrefix: String): ActorMaterializer =
```

Comment:
then it should also be added for Scala API:
```
def apply(materializerSettings: ActorMaterializerSettings, namePrefix: String)(implicit context: ActorRefFactory): ActorMaterializer =
```

Reply:
I have chaged it.

Reply:
I think I was wrong here. There is already one that can be used in Scala:
```
  def apply(materializerSettings: Option[ActorMaterializerSettings] = None, namePrefix: Option[String] = None)(
      implicit context: ActorRefFactory): ActorMaterializer 
```